src/artifactRegistry.ts

```diff
@@ -45,6 +56,69 @@ export class ArtifactRegistryDockerRegistryClient {
     this.repositoryFields = { project, location, repository };
   }
 
+  /**
+   * getGitCommitsBetweenTags
+   *
+   * @param {string} prevTag
+   *   The previous docker tag: of the following format: main---0013567-2024.04-g<githash>
+   * @param {string} nextTag
+   *  The next docker tag: main---0013567-2024.04-g<githash>
+   *
+   * @param {object} dockerImageRepository
+   *
+   * @returns {Promise} - The promise which resolves to an array of relevant commit strings.
+   */
+  async getGitCommitsBetweenTags({
+    prevTag,
+    nextTag,
+    dockerImageRepository,
+  }: {
+    prevTag: string;
+    nextTag: string;
+    dockerImageRepository: string;
+  }): Promise<string[]> {
+    core.info(
+      `running diff docker tags ${prevTag} ${nextTag} ${dockerImageRepository}`,
+    );
+    // Input is relatively trusted; this is largely to prevent mistaken uses
+    // like specifying a full Docker-style repository with slashes.
+    if (dockerImageRepository.includes('/')) {
+      throw Error('repository cannot contain a slash');
+    }
+
+    // Note: we don't need `listTagsAsync` (which is recommended) because we
+    // only care about the first element in the result array, which is the list of tags.
+    // https://github.com/googleapis/gax-nodejs/blob/main/client-libraries.md#auto-pagination
+    // These tags look like the following:
+    // {
+    //   "name":"projects/platform-cross-environment/locations/us-central1/repositories/platform-docker/packages/identity/tags/2022.02-278-g123456789",
+    //   "version":"projects/platform-cross-environment/locations/us-central1/repositories/platform-docker/packages/identity/versions/sha256:123abc456defg"
+    // }
+    const dockerTags = (
+      await this.client.listTags({
+        parent: this.client.pathTemplates.packagePathTemplate.render({
+          ...this.repositoryFields,
+          package: dockerImageRepository,
+        }),
+      })
+    )[0].filter((tag) => tag.version && tag.name);
```

Comment:
I think in practice these are actually always sent, right? And then this ends up pairing with the `as string`s  you have to do below? I think instead you can do a `.map(iTag => new Tag(iTag))` where `Tag` is imported from `@google-cloud/artifact-registry` and it will do what you want. (Google's generated protobuf types contain both an "input" version of each message where the fields are optional and a "class" version where they are not. This will mean that name/version could be empty strings of course... but in practice I don't think they are.)